core/config/project_settings.cpp

```diff
@@ -1342,6 +1342,73 @@ const HashMap<StringName, HashSet<StringName>> &ProjectSettings::get_scene_group
 	return scene_groups_cache;
 }
 
+int ProjectSettings::find_layer(LayerType p_type, const String &p_name) const {
```

Comment:
For example, I can imagine this entire implementation could be... restructured to use a Layer struct, along with the enums. Incoming psuedocode:
```C++
struct Layer {
    String basename;
    int32 limit;
}
```

But again, see above. I do not think these methods in **ProjectSettings** are a good idea, even if it makes sense at a glance.